network/server.go

```diff
@@ -650,3 +656,33 @@ func NewServer(
 
 	return &server
 }
+
+// CountConnections returns the current number of connections.
+func (s *Server) CountConnections() int {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return int(s.connections)
+}
+
+// Stop stops the server.
+func (s *Server) Stop() error {
```

Comment:
Unexport this function (make it internal) and remove it from the interface, as we don't want to stop the server without stopping the attached proxy instance, as the result would be unknown behavior. Optionally you can merge it with the `Shutdown` method. WDYT?

Reply:
@mostafa Thanks for your swift review, i pushed new commit with your comments